packages/integrations/src/unifi-controller/unifi-controller-integration.ts

```diff
@@ -0,0 +1,274 @@
+import { fetchWithTrustedCertificatesAsync } from "@homarr/certificates/server";
+
+import { Integration } from "../base/integration";
+import { IntegrationTestConnectionError } from "../base/test-connection-error";
+import type { NetworkControllerSummaryIntegration } from "../interfaces/network-controller-summary/network-controller-summary-integration";
+import type { NetworkControllerSummary } from "../interfaces/network-controller-summary/network-controller-summary-types";
+import { unifiSummaryResponseSchema } from "./unifi-controller-types";
+import { logger } from "@homarr/log";
+
+const udmpPrefix = "proxy/network";
+
+
+
+export class UnifiControllerIntegration extends Integration implements NetworkControllerSummaryIntegration {
+
+  private prefix: string | null | undefined;
+
+
+  static extractLoginTokenFromCookies(headers: Headers): string {
+    const cookies = headers.get("set-cookie") ?? "";
+    const loginToken = cookies
+      .split(";")
+      .find((cookie) => cookie.includes("TOKEN"));
+  
+    if (loginToken) {
+      return loginToken;
+    } else {
+      throw new Error("Login token not found in cookies");
+    }
+  }
+  
+  public async getNetworkSummaryAsync(): Promise<NetworkControllerSummary> {
+
+    if (!this.headers) {
+      await this.authenticateAndConstructSessionInHeaderAsync();
+    }
+
+    logger.debug("starting request");
+    //const endpoint = this.prefix === udmpPrefix ? "auth/login" : "login";
+    logger.debug("prefix: "+this.prefix); 
+
+    const requestUrl = this.url(`/${this.prefix}/api/stat/sites`);
+    logger.debug("requestUrl: "+requestUrl.toString()); 
+
+    const requestHeaders: Record<string, string> = { 
+      "Content-Type": "application/json",
+      ...this.headers,
+    };
+    if (this.csrfToken) {
+      requestHeaders["X-CSRF-TOKEN"] = this.csrfToken;
+    } 
+    logger.debug("requestHeaders: "+JSON.stringify(requestHeaders)); 
```

Comment:
Please never log any tokens or other confidential data. Remove this log statement. There is also not going to be much that will help you from this log message either.